modules/apps/frontend-view-state/frontend-view-state-service/src/main/java/com/liferay/frontend/view/state/internal/jaxrs/application/FrontendViewStateApplication.java

```diff
@@ -0,0 +1,147 @@
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Application;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.jaxrs.whiteboard.JaxrsWhiteboardConstants;
+
+/**
+ * @author Iván Zaera Avellón
+ */
+@Component(
+	property = {
+		JaxrsWhiteboardConstants.JAX_RS_APPLICATION_BASE + "=/frontend-view-state/app",
+		JaxrsWhiteboardConstants.JAX_RS_NAME + "=Liferay.Frontend.View.State",
+		"auth.verifier.auth.verifier.PortalSessionAuthVerifier.urls.includes=/*",
+		"auth.verifier.guest.allowed=true", "liferay.oauth2=false"
+	},
+	service = Application.class
+)
+public class FrontendViewStateApplication {
+
+	@GET
+	@Path("/active-view/{clayDataSetDisplayId}")
+	@Produces(MediaType.APPLICATION_JSON)
+	public Response getActiveView(
+		@PathParam("clayDataSetDisplayId") String clayDataSetDisplayId,
+		@QueryParam("plid") long plid,
+		@QueryParam("portletId") String portletId, @Context User user) {
+
+		try {
+			FVSActiveEntry fvsActiveEntry =
+				_fvsActiveEntryLocalService.fetchFVSActiveEntry(
+					user.getUserId(), clayDataSetDisplayId, plid, portletId);
+
+			if (fvsActiveEntry == null) {
+				return Response.status(
+					Response.Status.NOT_FOUND
+				).build();
+			}
+
+			FVSEntry fvsEntry = _fvsEntryLocalService.getFVSEntry(
+				fvsActiveEntry.getFvsEntryId());
+
+			return Response.ok(
+				_jsonFactory.createJSONObject(fvsEntry.getViewState()),
+				MediaType.APPLICATION_JSON
+			).build();
+		}
+		catch (Exception exception) {
+			_log.error("Unable to get active view", exception);
+		}
+
+		return Response.status(
+			Response.Status.NOT_FOUND
+		).build();
+	}
+
+	@Consumes(MediaType.APPLICATION_JSON)
+	@Path("/active-view/{clayDataSetDisplayId}/save")
+	@POST
+	public Response saveActiveView(
+		@PathParam("clayDataSetDisplayId") String clayDataSetDisplayId,
+		@QueryParam("plid") long plid,
+		@QueryParam("portletId") String portletId, @Context User user,
+		String viewState) {
+
+		try {
+			FVSActiveEntry fvsActiveEntry =
+				_fvsActiveEntryLocalService.fetchFVSActiveEntry(
+					user.getUserId(), clayDataSetDisplayId, plid, portletId);
+
+			if (fvsActiveEntry == null) {
+				FVSEntry fvsEntry = _fvsEntryLocalService.addFVSEntry(
+					user.getUserId(), viewState);
+
+				_fvsActiveEntryLocalService.addFVSActiveEntry(
+					user.getUserId(), fvsEntry.getFvsEntryId(),
+					clayDataSetDisplayId, plid, portletId);
+			}
+			else {
+				FVSEntry fvsEntry = _fvsEntryLocalService.getFVSEntry(
+					fvsActiveEntry.getFvsEntryId());
+
+				fvsEntry.setViewState(viewState);
+
+				_fvsEntryLocalService.updateFVSEntry(fvsEntry);
+			}
+
+			return Response.ok(
```

Comment:
In our APIs, we usually return the object with the response when it is created or updated